cmd/encode.go

```diff
@@ -30,28 +31,33 @@ import (
 // encodeCommand returns the command for `scte35 encode`
 func encodeCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "encode",
-		Short: "Encode a splice_info_section to binary",
+		Use:   "encode < filename",
+		Short: "Encode a splice_info_section to binary being provided from stdin",
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return fmt.Errorf("requires a splice_info_section")
-			}
-			if !strings.HasPrefix(args[0], "<") && !strings.HasPrefix(args[0], "{") {
-				return fmt.Errorf("splice_info_section must be in XML or JSON format")
+			if len(args) != 0 {
+				return fmt.Errorf("additional command line arguments are not needed")
 			}
 			return nil
 		},
 		Run: func(cmd *cobra.Command, args []string) {
-			bin := args[0]
+			reader := bufio.NewReader(os.Stdin)
+			var output []rune
```

Comment:
Would use a `string` here with `strings.HasPrefix` to detect an XML payload. This will save you from a panic if stdin is empty (empty file, for example)

Reply:
done